WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -431,6 +454,28 @@ private extension DashboardViewModel {
         }
     }
 
+    @MainActor
+    func updateHasOrdersStatus() async {
+        hasOrders = await loadHasOrdersStatus()
```

Comment:
This would require pulling-to-refresh the screen to make the analytics cards available again. Should we instead observe the storage and update `hasOrders` whenever there are changes to the storage? That would ensure that the dashboard screen is up-to-date when a store gets its first order.

Reply:
Added this in https://github.com/woocommerce/woocommerce-ios/pull/12521/commits/926beb88069473bc5179249aa9a0327031965f74

Based on our discussion about showing button, I removed `canShowShareStoreCard` since we don't have to check URL for the card visibility, and use the URL checking just for button showing.

This commit took a while because I ran into a situation where the card always appeared at the beginning when fetching wasn't done, but it shouldn't happen now.